interview-questions.md

```diff
@@ -6,62 +6,68 @@ First, without external resources. Challenge yourself to answer from memory as i
 
 Then, research the question to expand on your answer. Even if you feel you have answered the question completely, there is always something more to learn. Write your researched answer in your OWN WORDS.
 
-
 1. What is `this` in JavaScript?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer:
+'This', used in javascript, points to a declared variable within a class. We can then use that declared variable within a method which is also used in the class.
 
+Researched answer:
+It allows us to use the declared variable within a method also used in the class. If 'this' is used inside a method it points to the owner object. It can also be used on it's own when refering to the global object.
 
 2. What is React? Why would you use it?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer:
+React is a software created initially for Facebook, that contains a pre-built library for developers to use. The reason it was built was so web-pages could update and data could also be manipulated without the page refreshing. Refreshing a page is time and money.
 
+Researched answer:
+It is also component-based. The reason so many developers use it is because of how simple (and fun) it is to create interactive UIs.
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer:
+Render() is a required lifecycle method in a React class component.
 
+Researched answer:
+Any of the other methods are optional in a React class component, but you must include render().
 
 4. What is JSX? What is one notable difference between HTML and JSX?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer:
+JSX is a language that allows you to use html within a Javascript file. html can be written only in between a set of JSX brackets.
 
+Researched answer:
+It stands for JavaScript Syntax Extension, essentially writing in both HTML and Javascript. The main difference between HTML and JSX is that you must return only one parent element if you wnat it to compile.
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer:
+Yarn is a package manager for your code. that allows for developers to download packages directly into a project for use. There are a variety of different uses such as testing in jest or with a react project.
 
+Researched answer:
+Yarn is usually ran in terminal and brings over set packaged files for developers. Besides being a package mananger it is also a project manager.
+Developers can download plug-in features for better work flow and even share their code with others seemlessly.
```

Comment:
Yes, yarn will connect dependencies. The yarn.lock, package.json, and node_nodules will be updated.